includes/Core/Permissions/Permissions.php

```diff
@@ -428,9 +426,15 @@ private function check_view_splash_capability( $user_id ) {
 		if ( ! Feature_Flags::enabled( 'dashboardSharing' ) ) {
 			return array( self::AUTHENTICATE );
 		}
-		if ( in_array( 'do_not_allow', $this->check_view_shared_dashboard_capability( $user_id ), true ) ) {
+
+		if ( $this->is_shared_dashboard_splash_dismissed( $user_id ) ) {
+			return array( self::AUTHENTICATE );
+		}
+
+		if ( ! $this->user_has_shared_role( $user_id ) ) {
 			return array( self::AUTHENTICATE );
 		}
```

Comment:
In all of these cases, should the user maybe only be able to view the splash screen if they are not authenticated? This is probably not a big deal, but I believe right now they could still manually access the splash screen even if they were already authenticated, which I'm not sure is what we intend.

I guess since our plugin redirects you from splash to the dashboard in such a cases anyway, and since there is no security risk in such a case around accessing the splash screen again, we can just leave it like it is here. I just wanted to make sure we've considered this.

Reply:
Yes, that's how it is today. The user technically "can" view the splash but they're redirected to the dashboard if they can view it and vice-versa.